### PR TITLE
Fix dtw distance

### PR DESCRIPTION
**Pull request by FabienNugier**
Correct dtw_distance().

### CODE REVIEW

File: scifin/classifier/classifier.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def dtw_distance(ts1, ts2, window=None):
       Second time series.
     window : int
       Size of restrictive search window.
+    mode : str
+      Mode to choose from ('abs' or 'square').
```
Comment (godfrey-leung) [inline scifin/classifier/classifier.py]: add some words explaining what 'abs' and 'square' mean here?

File: scifin/classifier/classifier.py [hunk 2]
```diff
@@ -89,32 +91,42 @@ def dtw_distance(ts1, ts2, window=None):
...
     
     # Checks
-    try:
-        assert(ts1.type=='TimeSeries' and ts2.type=='TimeSeries')
-    except TypeError:
-        raise TypeError("Series have to be of type TimeSeries.")
-        
+    if not isinstance(ts1, ts.TimeSeries) and not isinstance(ts2, ts.TimeSeries):
+        raise AssertionError("Series have to be of type TimeSeries.")
+    if not isinstance(mode, str) and not mode in ['abs', 'square']:
```
Comment (godfrey-leung) [inline scifin/classifier/classifier.py]: for good coding practice, use `('abs', 'square')` instead so it is immutable (cannot be changed)

File: scifin/classifier/classifier.py [hunk 3]
```diff
@@ -89,32 +91,42 @@ def dtw_distance(ts1, ts2, window=None):
...
+            dtw[i,j] = cost + min(dtw[i-1,j], dtw[i,j-1], dtw[i-1,j-1])
+    if verbose:
+        print(dtw)
             
     # Return distance
-    return np.sqrt(dtw[N1, N2])
+    if mode=='abs':
+        return dtw[N1, N2]
+    elif mode=='square':
+        return np.sqrt(dtw[N1, N2])
```
Comment (godfrey-leung) [inline scifin/classifier/classifier.py:131]: so 'square' mean square root? I would prefer a different name than 'square' here

**Review by godfrey-leung — APPROVED** [review]

File: scifin/classifier/classifier.py (hunk 1 above)
Comment (FabienNugier) [reply scifin/classifier/classifier.py]: Indeed. Thanks.

File: scifin/classifier/classifier.py (hunk 2 above)
Comment (FabienNugier) [reply scifin/classifier/classifier.py]: Thanks for the tip.

File: scifin/classifier/classifier.py (hunk 3 above)
Comment (FabienNugier) [reply scifin/classifier/classifier.py:131]: No, it means the "cost" is a square distance. In fact, all the dtw matrix elements are square distances (and hence we need to take a sqrt at the end to still have a distance). I think from this point of view the name is appropriate, but I'm open to a better one.

**Pull request merged**